rasa_nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -146,13 +146,17 @@ def process(self, message, **kwargs):
                          "component is either not trained or "
                          "didn't receive enough training data")
         else:
-            bag = self.vect.transform([self._lemmatize(message)]).toarray()
+            bag = self.vect.transform(
+                [self._get_message_text(message)]
+            ).toarray()
             message.set("text_features", bag)
 
     @staticmethod
-    def _lemmatize(message):
-        if message.get("spacy_doc"):
+    def _get_message_text(message):
+        if message.get("spacy_doc"):  # if lemmatize is possible
             return ' '.join([t.lemma_ for t in message.get("spacy_doc")])
+        elif message.get("tokens"):  # if directly tokens is provided
```

Comment:
I think, it is better to add an additional flag to config, because if `tokenizer_whitespace` is present in the pipeline, extracting tokens is unnecessary.

Reply:
Well, in my opinion, additional flag will make code more complex which is unnecessary. if `tokenizer_whitespace` is present in the pipeline, it acted as a tokenizer, extracting tokens don't hurt the performance much but help code clean and easy to maintain.